web-marketplace/src/pages/CreateOrganization/CreateOrganization.tsx

```diff
@@ -0,0 +1,482 @@
+          twitterLink: values.twitterLink,
+          organizationId,
+          currentAccountId,
+        });
+
+        console.info('[CreateOrganization] dao created', {
+          organizationId: daoResult.organizationId,
+          daoAddress: daoResult.daoAddress,
+        });
+
+        daoAddressRef.current = daoResult.daoAddress;
+        organizationNameRef.current = values.name;
+        organizationIdRef.current = daoResult.organizationId;
+        setOrganizationProgressStep(daoResult.daoAddress, 1, values.name);
+
+        const payload: Record<string, unknown> = {
+          ...values,
+          dao: daoResult,
+        };
+
+        handleSaveNext(payload);
+      } catch (error) {
+        // Surface to the form error boundary for user feedback
+        console.error('Failed to create DAO', error);
+        throw error;
+      }
+    },
+    [createDao, handleSaveNext, hasUnfinishedOrganization, data],
+  );
+
+  const handlePrevClick = useCallback(() => {
+    if (activeStep === 0) return;
+    if (hasUnfinishedOrganization && daoAddressRef.current) {
+      const previousStep = Math.max(activeStep - 1, 1);
+      setOrganizationProgressStep(
+        daoAddressRef.current,
+        previousStep,
+        organizationNameRef.current,
+      );
+    }
+    handleBack();
+  }, [activeStep, handleBack, hasUnfinishedOrganization]);
+
+  const handleNextClick = useCallback(() => {
+    if (isCreating) return;
+
+    if (activeStep === 0) {
+      const form = document.getElementById(CREATE_ORG_FORM_ID) as
+        | HTMLFormElement
+        | undefined;
+
+      if (!form) return;
+
+      if (typeof form.requestSubmit === 'function') {
+        form.requestSubmit();
+      } else {
+        form.dispatchEvent(
+          new Event('submit', { bubbles: true, cancelable: true }),
+        );
+      }
+      return;
+    }
+
+    if (hasUnfinishedOrganization && daoAddressRef.current) {
+      if (isLastStep) {
+        removeOrganizationProgress(daoAddressRef.current);
+        daoAddressRef.current = undefined;
+        organizationNameRef.current = undefined;
+        setHasUnfinishedOrganization(false);
+      } else {
+        const nextStep = Math.max(activeStep + 1, 1);
+        setOrganizationProgressStep(
+          daoAddressRef.current,
+          nextStep,
+          organizationNameRef.current,
+        );
+      }
+    }
+
+    handleSaveNext({});
+  }, [
+    activeStep,
+    handleSaveNext,
+    hasUnfinishedOrganization,
+    isCreating,
+    isLastStep,
+  ]);
+
+  return (
+    <>
+      {activeStep === 0 && (
+        <OrganizationProfileStep
+          formId={CREATE_ORG_FORM_ID}
+          // Remount when we set transferred values to ensure RHF defaultValues apply
+          key={JSON.stringify(orgProfileInitialValues)}
+          initialValues={orgProfileInitialValues}
+          onSaved={handleOrganizationProfileSaved}
+        />
+      )}
+      <TransferProfileModal
```

Comment:
this should be in `OrganizationProfileStep` we do not need this modal for other steps